pkg/backporter/backporter_test.go

```diff
@@ -199,15 +182,137 @@ func TestGetClonesHandler(t *testing.T) {
 			}
-			if resp := rr.Body.String(); resp != tc.results.htmlPage {
-				t.Errorf("Response differs from expected by: %s", diff.StringDiff(resp, tc.results.htmlPage))
+			var buf bytes.Buffer
+			if err := tc.results.tmplt.Execute(&buf, tc.results.data); err != nil {
+				t.Errorf("Unable to render template: %v", err)
+			}
+			subPage := buf.String()
+			var expResponse string
+			if tc.results.statusCode == http.StatusOK {
+				expResponse = fmt.Sprintf(htmlPageStart, "Clones") + subPage + htmlPageEnd
+			} else {
+				expResponse = fmt.Sprintf(htmlPageStart, "Not Found") + subPage + htmlPageEnd
+			}
+			if resp := rr.Body.String(); resp != expResponse {
+				t.Errorf("Response differs from expected by: %s", diff.StringDiff(resp, expResponse))
 			}
 		})
 	}
 
 }
+
+func TestCreateCloneHandler(t *testing.T) {
+	fake := &bugzilla.Fake{}
+	fake.Bugs = map[int]bugzilla.Bug{}
+	fake.BugComments = map[int][]bugzilla.Comment{}
+
+	bug1Create := &bugzilla.BugCreate{
+		AssignedTo: "UnitTest",
+		Summary:    "Sample bug to test implementation of clones handler",
+	}
+	bug1ID, err := fake.CreateBug(bug1Create)
+	if err != nil {
+		t.Errorf("Error creating bug: %v", err)
+	}
+	clonedRelease := "4.4.z"
+	prunedReleaseSet := sets.NewString(allTargetVersions.List()...)
+	prunedReleaseSet.Delete(clonedRelease)
+
+	bug1, err := fake.GetBug(bug1ID)
+	if err != nil {
+		t.Errorf("Error getting bug details from Fake.")
+	}
+	expectedCloneID := bug1ID + 1
+	testcases := []struct {
+		name    string
+		params  map[string]string
+		results ResCheck
+	}{
+		{
+			"valid_parameters",
+			map[string]string{
+				"ID":      strconv.Itoa(bug1ID),
+				"release": clonedRelease,
+			},
+			ResCheck{
+				http.StatusOK,
+				ClonesTemplateData{
+					bug1,
+					[]*bugzilla.Bug{},
+					bug1,
+					nil,
+					prunedReleaseSet.List(),
+					expectedCloneID,
+				},
+				clonesTemplate,
+			},
+		},
+		{
+			"bad_params",
+			map[string]string{
+				"ID":      "1000",
+				"release": "",
+			},
+			ResCheck{
+				http.StatusNotFound,
+				"Unable to fetch bug details- Bug#1000 : bug not registered in the fake",
+				errorTemplate,
+			},
+		},
+	}
+	for _, tc := range testcases {
+		t.Run(tc.name, func(t *testing.T) {
+			formData := url.Values{}
+			formData.Set("ID", tc.params["ID"])
+			formData.Add("release", tc.params["release"])
+			req, err := http.NewRequest("POST", "/clones/create", bytes.NewBufferString(formData.Encode()))
+			req.Header.Set("Content-Type", "application/x-www-form-urlencoded; param=value")
+			if err != nil {
+				t.Fatal(err)
+			}
+			rr := httptest.NewRecorder()
+			handler := unwrapper(CreateCloneHandler(fake, allTargetVersions))
+			handler.ServeHTTP(rr, req)
+			if status := rr.Code; status != tc.results.statusCode {
+				t.Errorf("testcase '%v' failed: clonebug returned wrong status code - got %v, want %v", tc, status, tc.results.statusCode)
```

Comment:
`Fatalf` IMHO

Reply:
But since we evaluate multiple testcases in this loop, wouldn't we want it to just register the error and continue in this case?

Reply:
So the way this works is that everytime you call `t.Run` you start a new (sub-) test. If you then call `t.Fatalf` you only stop the execution of that subtest, not of the parent or the silblings in the loop.

Reply:
But in this case, I'm pretty sure I want to also check the response body even if the headers differ. Why do you feel otherwise?

Reply:
Because the header indicates which codepath we took (e.G. success/failure) and if that is wrong already we can very strongly assume that the response body will be wrong as well

Reply:
True... but it could also just be that the header passed is a 404 when what was expected was a 500 and that doesn't necessarily call for aborting the tc

Reply:
its not a strong opinion one way or another